src/utils/fs.rs

```diff
@@ -80,6 +80,77 @@ pub fn nice_directory_display(os_str: impl AsRef<OsStr>) -> Cow<'static, str> {
     }
 }
 
+/// Try to detect the file extension by looking for known magic strings
+pub fn try_infer(path: &Path) -> Option<Extension> {
+    fn is_zip(buf: &[u8]) -> bool {
+        buf.len() > 3
+            && buf[0] == 0x50
+            && buf[1] == 0x4B
+            && (buf[2] == 0x3 || buf[2] == 0x5 || buf[2] == 0x7)
+            && (buf[3] == 0x4 || buf[3] == 0x6 || buf[3] == 0x8)
+    }
+    fn is_tar(buf: &[u8]) -> bool {
+        buf.len() > 261
+            && buf[257] == 0x75
+            && buf[258] == 0x73
+            && buf[259] == 0x74
+            && buf[260] == 0x61
+            && buf[261] == 0x72
+    }
```

Comment:
I wonder, what are the differences between doing this and using a crate like `infer`?

Reply:
It is the same as infer (because I copied code from it), except infer is missing is_lz4 and is_zst

Reply:
Infer: https://github.com/bojand/infer/blob/3221954081587a8dbaee83f0b98a1bee49c54193/src/lib.rs#L236

Reply:
Nice, I'll be copying the LICENSE copyright notice.